core/opendaq/component/include/opendaq/component_impl.h

```diff
@@ -270,6 +270,9 @@ ErrCode ComponentImpl<Intf, Intfs...>::setActive(Bool active)
     if (this->frozen)
         return OPENDAQ_ERR_FROZEN;
 
+    if (this->isComponentRemoved)
```

Comment:
I know locking is very incosistent at the moment, but IMHO this should pushed down behind the mutext lock.